### PR TITLE
OPT: avoid calling ds.subdatasets per each submodule path

### PR DESCRIPTION
**Pull request by yarikoptic**
it is not cheap. Best to get them all first at once

note: I didn't try it yet and don't know if tests cover this

### CODE REVIEW

File: tools/backups2datalad/adataset.py [hunk 1]
```diff
@@ -268,15 +270,16 @@ async def get_stats(
...
             path = Path(filestat.path)
             if not is_meta_file(path.parts[0], dandiset=True):
                 if filestat.type is ObjectType.COMMIT:
                     # this zarr should not be present locally as a submodule
-                    # so we should get its id from its information in submodules
-                    sub_info = await self.get_subdatasets(path=path)
-                    assert len(sub_info) == 1  # must be known
-                    zarr_id = Path(sub_info[0]["gitmodule_url"]).name
+                    # so we should get its id from its information in submodules.
+                    sub_info = subdatasets[str(path.absolute())]
```
Comment (jwodder) [inline tools/backups2datalad/adataset.py]: `path` is relative to `self.pathobj` (the path to the root of the dataset), not to the current directory, so `path.absolute()` won't give the correct result.

**Review by jwodder — CHANGES_REQUESTED** [review]

File: tools/backups2datalad/adataset.py (hunk 1 above)
Comment (yarikoptic) [inline tools/backups2datalad/adataset.py]: so we need smth like
```suggestion
                    sub_info = subdatasets[str(self.pathobj / path)]
```
?

Comment (jwodder) [reply tools/backups2datalad/adataset.py]: @yarikoptic Yes.

**Comment by yarikoptic** [comment]
concern was addressed and tests pass now , let's proceed

**Pull request merged**